### PR TITLE
Fix Loss Function to Improve Model Convergence for `AutoEncoder`

### LINKED ISSUE #1455: [BUG]:  DFencoder Model struggles to converge when it is trained solely with numerical features
### Version

23.11

### Which installation method(s) does this occur on?

Kubernetes

### Describe the bug.

The DFencoder Model struggles to converge when it is trained solely with numerical features, as evidenced by the training and validation loss remaining consistent and not decreasing. However, when a new categorical feature is introduced to the dataset, the model begins to converge, resulting in a noticeable reduction in both training and validation loss.

### Minimum reproducible example

```shell
Use IRIS dataset.
1. Use only numerical features, train dfencoder model observe training and validation loss.
2. Add constant string feature to dataset, train dfencoder model observe training and validation loss
```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

Attaching training and validation loss with both scenario

</pre></details>
error

### Full env printout

<details><summary>Click here to see environment details</summary><pre>

 [Paste the results of print_env.sh here, it will be hidden by default]

</pre></details>

### Other/Misc.

_No response_

### Code of Conduct

- [X] I agree to follow Morpheus' Code of Conduct
- [X] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by hsin-c**
## Description
This PR addresses an issue in the dfencoder model related to its convergence behavior. Previously, the model exhibited difficulty in converging when trained exclusively with numerical features. 
This PR fixes the way different loss types are combined in the model's loss function to ensure that backpropagation works correctly.

Note: This may alter the exact values resulting from calling `fit()` on the model. Before, categorical features were weighted much higher than binary or numerical categories (all numerical features shared a combined weight of 1, all binaries features shared a combined weight of 1, and each categorical feature had a weight of 1). Now all features are weighted equally which may impact the trained weights.

Closes #1455

### CODE REVIEW

**Review by mdemoret-nv — APPROVED** [review]
Reviewed the changes and have made a few commits to improve the readability and remove redundant calculations. The loss now equally weights all types of features. Once CI is passing and @hsin-c  is good with the changes, we can merge.

**Review by mdemoret-nv — APPROVED** [review]
Thanks for including the use of `.sum()`. Looks good now.
